polygeist/lib/polygeist/Passes/ConvertPolygeistToLLVM.cpp

```diff
@@ -735,6 +798,12 @@ struct ConvertPolygeistToLLVMPass
       patterns
           .add<LLVMOpLowering, GlobalOpTypeConversion, ReturnOpTypeConversion>(
               converter);
+
+      // TODO: This is a temporal solution. In the future, we might want to
+      // handle GPUDialect lowering by extending the GpuToLLVMConversionPass.
```

Comment:
What is missing from the current `GpuToLLVMConversionPass`? Does it handle `gpu.return`?
Should we add `GpuToLLVMConversion` as one of the patterns anyways?

Reply:
Every GPU operation we're generating (including `gpu.return`) is being handled. My original plans were to add the `GpuToLLVMConversion`, but that's just discarding the GPU code and representing kernel launches as a sequence of library calls (see [here](https://github.com/intel/llvm/blob/eea3f8b6c3fa17fac50b1d6384e965cba2cc62f6/mlir/lib/Conversion/GPUCommon/GPUToLLVMConversion.cpp#L725-L739)):

```
 725 // Emits LLVM IR to launch a kernel function. Expects the module that contains                                                                                                                                 
 726 // the compiled kernel function as a cubin in the 'nvvm.cubin' attribute, or a                                                                                                                                 
 727 // hsaco in the 'rocdl.hsaco' attribute of the kernel function in the IR.                                                                                                                                      
 728 //                                                                                                                                                                                                             
 729 // %0 = call %binarygetter                                                                                                                                                                                     
 730 // %1 = call %moduleLoad(%0)                                                                                                                                                                                   
 731 // %2 = <see generateKernelNameConstant>                                                                                                                                                                       
 732 // %3 = call %moduleGetFunction(%1, %2)                                                                                                                                                                        
 733 // %4 = call %streamCreate()                                                                                                                                                                                   
 734 // %5 = <see generateParamsArray>                                                                                                                                                                              
 735 // call %launchKernel(%3, <launchOp operands 0..5>, 0, %4, %5, nullptr)                                                                                                                                        
 736 // call %streamSynchronize(%4)                                                                                                                                                                                 
 737 // call %streamDestroy(%4)                                                                                                                                                                                     
 738 // call %moduleUnload(%1) 
```

As I say, this is a temporary solution to lower to LLVM. It will work for now, but we have to decide on how to represent both modules.